ProcessLib/Parameter/PropertyIndexParameter.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_PROPERTYINDEXPARAMETER_H_
+#define PROCESSLIB_PROPERTYINDEXPARAMETER_H_
+
+#include "Parameter.h"
+
+#include "MeshLib/PropertyVector.h"
+
+namespace MeshLib
+{
+template <typename T>
+class PropertyVector;
+}  // MeshLib
+
+namespace ProcessLib
+{
+
+/// A parameter class looking for values from index in a property vector.
+/// This class can be used for material ID depependent parameters.
+template <typename T, MeshLib::MeshItemType MeshItemType>
+struct PropertyIndexParameter final
```

Comment:
I don't see importance of it.  Other parameters like MeshNodeParameter do not have the keyword.